js/ez-field-edit.js

```diff
@@ -0,0 +1,94 @@
+(function () {
+    /**
+    *  `<ez-field-edit>` is supposed to wrap one or several HTML5 inputs. It tracks user input,
+    *  and if an input is invalid, sets its own `invalid` property to true. It will also add an
+    * `ez-invalid` class on the related label. This class will be removed once the input gets valid.
+    * @polymerElement
+    * @demo demo/ez-field-edit.html
+    */
+    class FieldEdit extends Polymer.Element {
+        static get is() {
+            return 'ez-field-edit';
+        }
+
+        static get properties() {
+            return {
+                'invalid': {
+                    type: Boolean,
+                    value: false,
+                    reflectToAttribute: true,
+                },
+            };
+        }
+
+        constructor() {
+            super();
+            this.addEventListener('input', this._checkValidity.bind(this));
+        }
+
+        /**
+         * Checks whether the inputs are valid or not and set the `invalid` property
+         * accordingly.
+         * Also set `ez-invalid` class on labels connected to invalid inputs.
+         */
+        _checkValidity() {
+            let valid;
+
+            this.querySelectorAll('input, select, textarea').forEach((formElement) => {
+                valid = formElement.validity.valid;
+
+                this.querySelectorAll('label').forEach((labelElement) => {
+                    if (!valid) {
+                        this._commonAncestor(labelElement, formElement).classList.add('ez-invalid');
+                    } else {
+                        this._commonAncestor(labelElement, formElement).classList.remove('ez-invalid');
+                    }
+                });
+            });
+            this.invalid = !valid;
+        }
+
+        /**
+         * Returns the parentNodes of the given node untill <ez-field-edit>.
+         *
+         * @param {HTMLElement} node
+         * @return {Array}
+         */
+        _parents(node) {
+            const nodes = [];
+
+            for (; node; node = node.parentNode) {
+                nodes.unshift(node);
+                if (node === this) {
+                    break;
+                }
+            }
+            return nodes;
+        }
+
+        /**
+         * Returns the lowest common ancestor between the two given nodes.
+         *
+         * @param {HTMLElement} node1
+         * @param {HTMLElement} node2
+         * @return {HTMLElement}
+         */
+        _commonAncestor(node1, node2) {
+            const parents1 = this._parents(node1);
+            const parents2 = this._parents(node2);
+
+            for (let i = 0; i < parents1.length; i++) {
+                if (parents1[i] != parents2[i]) {
```

Comment:
I have some doubts if this is really working if `node1` and `node2` have a different depth. You should improve the test various cases.